File: frontend/src/types/taxStatutory.ts
export interface StatutoryRule {
  id: number;
  country: string;
  rule_code: string;
  name: string;
  employee_rate: number;
  employer_rate: number;
  wage_ceiling?: number;
  monthly_cap?: number;
  eligibility_threshold?: number;
  rule_version: string;
  effective_date: string;
  is_active: boolean;
}

export interface TaxSlab {
  id: number;
  from_income: number;
  to_income?: number;
  tax_rate: number;
}

export interface TaxRule {
  id: number;
  country: string;
  financial_year: string;
  regime_name: string;
  standard_deduction: number;
  cess_rate: number;
  rule_version: string;
  effective_date: string;
  is_active: boolean;
  tax_slabs: TaxSlab[];
}

export interface TDSEvaluationResult {
  annual_gross: number;
  standard_deduction: number;
  taxable_annual: number;
  tax_before_cess: number;
  cess_amount: number;
  total_annual_tax: number;
  monthly_tds: number;
  regime_name: string;
  rule_version: string;
}

File: backend/app/schemas/tax_statutory.py
from datetime import date
from typing import List, Optional
from pydantic import BaseModel, Field


class StatutoryRuleCreate(BaseModel):
    country: str = "India"
    rule_code: str = Field(..., example="IN_PF")
    name: str = Field(..., example="Employee Provident Fund")
    employee_rate: float = Field(..., example=0.1200)
    employer_rate: float = Field(..., example=0.1200)
    wage_ceiling: Optional[float] = 15000.0
    monthly_cap: Optional[float] = 1800.0
    eligibility_threshold: Optional[float] = None
    rule_version: str = "v2024.1"
    effective_date: date
    is_active: bool = True


class StatutoryRuleResponse(StatutoryRuleCreate):
    id: int

    model_config = {"from_attributes": True}


class TaxSlabItem(BaseModel):
    from_income: float
    to_income: Optional[float] = None
    tax_rate: float


class TaxRuleCreate(BaseModel):
    country: str = "India"
    financial_year: str = "2024-2025"
    regime_name: str = "New Regime"
    standard_deduction: float = 75000.0
    cess_rate: float = 0.0400
    rule_version: str = "v2024.1"
    effective_date: date
    is_active: bool = True
    slabs: List[TaxSlabItem] = []


class TaxSlabResponse(BaseModel):
    id: int
    from_income: float
    to_income: Optional[float] = None
    tax_rate: float

    model_config = {"from_attributes": True}


class TaxRuleResponse(BaseModel):
    id: int
    country: str
    financial_year: str
    regime_name: str
    standard_deduction: float
    cess_rate: float
    rule_version: str
    effective_date: date
    is_active: bool
    tax_slabs: List[TaxSlabResponse] = []

    model_config = {"from_attributes": True}


class EvaluateTDSRequest(BaseModel):
    gross_monthly_salary: float
    regime_name: str = "New Regime"
